hs_core/views/resource_folder_hierarchy.py

```diff
@@ -27,21 +27,21 @@ def data_store_structure(request):
     """
     res_id = request.POST.get('res_id', None)
     if res_id is None:
-        return HttpResponse(content_type='application/json', status=400)
+        return HttpResponse('Bad request - resource id is not included', status=400)

```

Comment:
@pkdash I googled and did not find any rule against returning a response with a content type other than JSON to an AJAX request, since our current client AJAX code error handling for folders work all assumes the response has a text/html content type, hence the changes I made for errors to be handled by the client properly. What I meant by "HttpResponse object can be directly used without explictly setting it, which is what has been done across our code base" was to explain the fact that I did not explicitly set the content type to be text/html but rather let the default to take effect which is what has been done across our code base. We only return error messages via a content type of text/html, but return a JSON content type in the response if it succeeds, so we can expand the content to return to client as we wish when it succeeds, and I don't anticipate anything extra to return except for error message and status code when it fails, so I think what is done currently should be just fine. If you could point me to any rule or best practice against returning a httpResponse with txt/html content type to an AJAX call when it errors out, I would take the trouble to change the server responses in many places and client error handling in many places, otherwise, I think it is fine to do what we do now which is working as expected.